addons/website/static/src/interactions/cookies/cookies_approval.js

```diff
@@ -0,0 +1,100 @@
+import { registry } from "@web/core/registry";
+import { MEDIAS_BREAKPOINTS, SIZES } from "@web/core/ui/ui_service";
+import { renderToElement } from "@web/core/utils/render";
+import { Interaction } from "@website/core/interaction";
+
+export class CookiesApproval extends Interaction {
+    static selector = "[data-need-cookies-approval]";
+
+    dynamicSelectors = {
+        "_iframe": () => this.iframeEl,
+    };
+    dynamicContent = {
+        "_iframe:t-att-class": () => ({ "d-none": this.warningShown }),
+    };
+
+    setup() {
+        this.iframeEl = this.el.tagName === "IFRAME" ? this.el : this.el.querySelector("iframe");
+        this.warningShown = true;
+    }
+
+    start() {
+        if (this.iframeEl) {
+            // The warningEl is not queried from the root, because there might
+            // be more than one iframe inside it.
+            this.optionalCookiesWarningEl = this.iframeEl.nextElementSibling
+                ?.classList.contains("o_no_optional_cookie")
+                ? this.iframeEl.nextElementSibling
+                : null;
+            if (!this.optionalCookiesWarningEl) {
+                this.addOptionalCookiesWarning();
+            }
+        }
+        this.addListener(
+            this.services.website_cookies.bus,
+            "add_cookies_warning",
+            this.onAddCookiesWarning
+        );
+    }
+
+    destroy() {
+        if (this.warningShown) {
+            this.removeOptionalCookiesWarning();
+        }
+    }
+
+    /**
+     * Adds a warning in place of the iframe. On click, shows the cookies bar if
+     * it was hidden.
+     */
+    addOptionalCookiesWarning() {
+        this.optionalCookiesWarningEl = renderToElement("website.cookiesWarning", {
+            extraStyle: this.iframeEl.parentElement.classList.contains("media_iframe_video")
+                ? `aspect-ratio: 16/9; max-width: ${MEDIAS_BREAKPOINTS[SIZES.SM].maxWidth}px;`
+                : "",
+            extraClasses: getComputedStyle(this.iframeEl.parentElement).position === "absolute"
+                ? "" : "my-3",
+        });
+        this.insert(this.optionalCookiesWarningEl, this.iframeEl, "afterend");
+
+        this.warningShown = true;
+        this.addListener(this.optionalCookiesWarningEl, "click", () => {
+            this.services.website_cookies.bus.trigger("show_cookies_bar");
+        });
+        this.addListener(
+            document,
+            "optionalCookiesAccepted",
+            this.removeOptionalCookiesWarning,
+            { once: true }
+        );
```

Comment:
What about defining an interaction for this, and starting it on the inserted element ?